walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -46,53 +46,40 @@
         return Arrays.asList(TimeRange.WHOLE_DAY);
     }
 
-    //Change to ArrayList and sort timeRanges by starting time 
     List<TimeRange> timeRanges = new ArrayList<TimeRange>(timeRangesSet);
+    addDummyEvents(timeRanges);
     Collections.sort(timeRanges, TimeRange.ORDER_BY_START);
 
     Collection<TimeRange> availTimes = new ArrayList<TimeRange>(); //Time ranges that are available for the meeting request
-    int startOfAvail = TimeRange.START_OF_DAY; //The start of an available time range
+    
+    TimeRange lastTimeRange = timeRanges.get(0); //Last time range that has been accounted for from timeRanges
+    TimeRange nextTimeRange; //Next time range that needs to be accounted for
+    int timeRangesCtr = 1; 
+        //Index of current position in array list "timeRanges", 
+        //start at 1 since timeRange at index 0 is dummy and 
+        //has already been accounted for
+    int startOfAvail = lastTimeRange.end();
     int endOfAvail; 
-    int timeRangesCtr = 0; //Index of current position in array list "timeRanges"
-    TimeRange lastTimeRange = null; //Last time range that has been accounted for from timeRanges
-    TimeRange nextTimeRange = timeRanges.get(0);; //Next time range that needs to be accounted for
-
-    if(nextTimeRange.start() == TimeRange.START_OF_DAY){
-        //If earliest event starts at 00:00 then update startOfAvail 
-        //to be the end time of the first event
-        startOfAvail = nextTimeRange.end();
-
-        //First event is accounted for, so update lastTimeRange
-        lastTimeRange = nextTimeRange;
-        timeRangesCtr++;
-    } 
 
     while(startOfAvail < TimeRange.END_OF_DAY && timeRangesCtr < timeRanges.size()){ 
         nextTimeRange = timeRanges.get(timeRangesCtr);
-
-        if((null != lastTimeRange) && (lastTimeRange.overlaps(nextTimeRange))) {
-            startOfAvail = nextTimeRange.end();
-            timeRangesCtr++;
-        } else if ((null != lastTimeRange) && (lastTimeRange.contains(nextTimeRange))){
+        if ((lastTimeRange.contains(nextTimeRange))){
```

Comment:
nit: You have an extra set of parentheses here that don't help and sort of make it harder to read. Same below.